compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ClosureGenerator.java

```diff
@@ -569,11 +599,42 @@ private BInvokableSymbol createSimpleVariable(BLangFunction function, BLangLambd
         BLangSimpleVariableDef variableDef = ASTBuilderUtil.createVariableDef(function.pos);
         variableDef.var = simpleVariable;
         variableDef.setBType(type);
+        if (isAnnotationClosure) {
+            annotationClosureReferences.add(variableDef);
+            return varSymbol;
+        }
         queue.add(variableDef);
+        return varSymbol;
+    }
 
+    public BVarSymbol createSimpleVariable(BInvokableSymbol invokableSymbol, boolean isAnnotationClosure) {
```

Comment:
Why is this public?

Reply:
Change to private(https://github.com/ballerina-platform/ballerina-lang/pull/39349)